### PR TITLE
First commits

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

This pull request provides a Terraform configuration to provide a singular AWS S3 bucket to store AWS Lambda deployment artifacts in [CyHy] environments.

## 💭 Motivation and context ##

As we have added AWS Lambdas to the [CyHy] configuration they each have gotten their own S3 bucket to store deployment artifacts. This will, in my opinion, only get messier as we add more Lambdas to the environment configuration. I believe that a single bucket to store deployment artifacts for a given [CyHy] workspace is both more manageable and makes more sense in the long run. This will allow us to remove the existing artifact buckets for the [assessment-data-import](https://github.com/cisagov/assessment-data-import-terraform/blob/develop/lambda_bucket.tf) and [findings-data-import](https://github.com/cisagov/findings-data-import-terraform/blob/develop/lambda_bucket.tf) Lambadas.

> [!NOTE]
> Once this PR is approved and merged I will deploy it to production and begin work on dismantling the aforementioned bucket configurations . This deploy should not be problematic because this configuration should have no conflicts with existing infrastructure.

## 🧪 Testing ##

Automated tests pass.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

[CyHy]: https://github.com/cisagov/cyhy_amis

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,46 +2,74 @@
...
 
-This is a generic skeleton project that can be used to quickly get a
-new [cisagov](https://github.com/cisagov) [Terraform
-module](https://www.terraform.io/docs/modules/index.html) GitHub
-repository started.  This skeleton project contains [licensing
-information](LICENSE), as well as [pre-commit
-hooks](https://pre-commit.com) and
-[GitHub Actions](https://github.com/features/actions) configurations
-appropriate for the major languages that we use.
+This project creats an AWS S3 bucket to store the deployment artifacts for any
```
Comment (dav3r) [inline README.md]: Fix typo:
```suggestion
This project creates an AWS S3 bucket to store the deployment artifacts for any
```

File: README.md [hunk 2]
```diff
@@ -2,46 +2,74 @@
...
+  [configured](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html)
+  for the appropriate account on your system.
+- An accessible AWS S3 bucket to store Terraform state
+  (specified in [`backend.tf`](backend.tf)).
+- An accessible AWS DynamoDB database to store the Terraform state lock
+  (specified in [`backend.tf`](backend.tf)).
+
+## Customizing Your Environment ##
+
+Create a terraform variables file to be used for your environment (e.g.
```
Comment (dav3r) [inline README.md]: Proper capitalization:
```suggestion
Create a Terraform variables file to be used for your environment (e.g.
```

**Review by dav3r — APPROVED** [review]
I completely agree with you that this is the way.  I noted a couple of tiny thangs, but otherwise, LGTM. 👍

File: README.md (hunk 2 above)
Comment (mcdonnnj) [reply README.md]: Ack that's what I get for blindly copying and pasting from another project.

**Review by jsf9k — APPROVED** [review]

**Review by jasonodoom — APPROVED** [review]

**Comment by mcdonnnj** [comment]
@dav3r and @jsf9k I force-pushed after updating the `lambda_artifact_s3_bucket` to `lambda_artifacts_s3_bucket` to better represent that it will store multiple artifacts (which aligns with the bucket name etc.). I managed to forget to pull down from `origin` before rebasing and had to manually recreate the commit with @dav3r's suggestions. Just wanted a double-check to make sure I didn't botch something else.

**Review by dav3r — APPROVED** [review]
Still LGTM 👍

**Review by jsf9k — COMMENTED** [review]
Should we also update the name of the `lambda_artifact_bucket.tf` file?

**Comment by mcdonnnj** [comment]
> Should we also update the name of the `lambda_artifact_bucket.tf` file?

@dav3r @jsf9k Please see [2426069](https://github.com/cisagov/cyhy-lambda-bucket-terraform/pull/3/commits/2426069e9c03c3f616f663f6575acb4399447017).

**Review by jsf9k — APPROVED** [review]

**Pull request merged**